app/src/main/java/fragments/MainFragment.kt

```diff
@@ -1,59 +1,99 @@
-package fragments;
+package fragments
 
-import android.animation.AnimatorInflater
 import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import android.widget.LinearLayout
-import androidx.core.view.children
+import androidx.recyclerview.widget.RecyclerView
+import data.Sound
+import adapter.SoundAdapter
+import android.animation.*
+import android.content.res.Configuration
+import android.util.Log
+import android.widget.Button
+import androidx.recyclerview.widget.GridLayoutManager
 import lufra.youpidapp.MainActivity
 import lufra.youpidapp.R
 
 class MainFragment: MyFragment() {
     private lateinit var context: MainActivity
+    private lateinit var recyclerView: RecyclerView
+    private lateinit var viewAdapter: SoundAdapter
+    private lateinit var viewLayoutManager: GridLayoutManager
+    private var viewColumnCount = 1
     override var TAG: String = "=====MAINFRAGMENT====="
 
+    private val animCleanup = object: SoundAdapter.CleanupAnimationListener {
+        override fun onUnbind(soundViewHolder: SoundAdapter.SoundViewHolder) {
+            soundViewHolder.button.setBackgroundColor(resources.getColor(R.color.colorPrimary, null))
+        }
+    }
+
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                               savedInstanceState: Bundle?): View? {
         super.onCreateView(inflater, container, savedInstanceState)
         context = activity as MainActivity
-        return inflater.inflate(R.layout.fragment_main, container, false)
+        val view = inflater.inflate(R.layout.fragment_main, container, false)
+        // Not sure if the following should be put in onCreateView or in onActivityCreated
+        viewColumnCount = resources.getInteger(R.integer.grid_column_count)
+        viewLayoutManager = GridLayoutManager(view.context, viewColumnCount)
+        viewAdapter = SoundAdapter(Sound.ALL_SOUNDS_STR.mapIndexed { index, s ->
+            Sound(index, s)
+        }, object: SoundAdapter.SoundClickedListener {
+            override fun onSoundClicked(soundViewHolder: SoundAdapter.SoundViewHolder) {
+                val sound = soundViewHolder.sound!!
+                val soundName = sound.name
+                val duration = context.discotheque.play(soundName)
+                playWtfAnimator(sound, duration.toLong())
+                Log.d(TAG, "Clicked, $soundViewHolder")
+            }
```

Comment:
why this object is not made like the animCleanup?  To be consistent, shouldn't it be better if we put it as a val like the other object?

Reply:
Yep, it would be cleaner